dpytools/http/upload/token_auth.py

```diff
@@ -0,0 +1,123 @@
+import os
+from datetime import datetime, timedelta
+from typing import Dict
+
+from dpytools.http.base import BaseHttpClient
+from dpytools.logging.logger import DpLogger
+
+logger = DpLogger("dpytools")
+
+
+class TokenAuth(BaseHttpClient):
+    def __init__(self, backoff_max=30):
+        super().__init__(backoff_max=backoff_max)
+        # RE auth, there'd two modes
+        # 1. Service account mode
+        # 2. User account mode
+        # The headers used are slightly different and
+        # user account mode required refreshing the auth
+        # token before a 15 minute timeout happens
+        # (service account auth doesn't time out)
+
+        self.service_token = os.environ.get("SERVICE_TOKEN_FOR_UPLOAD", None)
+        if self.service_token is None:
+            self.set_user_tokens()
+
+    def set_user_tokens(self):
+        """
+        When using user auth we need to use florence username and password
+        to create a florence token.
+
+        We also need to get the refresh token so we can extend the token
+        lifespan beyond the 15 minute timeout where necessary.
+        """
+
+        self.florence_user = os.environ.get("FLORENCE_USER", None)
+        self.florence_password = os.environ.get("FLORENCE_PASSWORD", None)
+        self.identity_api_url = os.environ.get("IDENTITY_API_URL", None)
+
+        assert (
```

Comment:
Could these assertions on set_user_tokens be tested? They are small checks but I think they are somewhat significant. What do you think?